spring-native/src/main/java/org/springframework/nativex/substitutions/data/Target_CustomRepositoryImplementationDetector.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.nativex.substitutions.data;
+
+import java.io.IOException;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.oracle.svm.core.annotate.Substitute;
+import com.oracle.svm.core.annotate.TargetClass;
+import org.springframework.beans.factory.BeanDefinitionStoreException;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.ScannedGenericBeanDefinition;
+import org.springframework.context.index.CandidateComponentsIndex;
+import org.springframework.context.index.CandidateComponentsIndexLoader;
+import org.springframework.core.type.classreading.MetadataReader;
+import org.springframework.data.repository.config.ImplementationDetectionConfiguration;
+import org.springframework.nativex.substitutions.OnlyIfPresent;
+import org.springframework.stereotype.Component;
+
+
+@TargetClass(className = "org.springframework.data.repository.config.CustomRepositoryImplementationDetector", onlyWith = {OnlyIfPresent.class})
+public final class Target_CustomRepositoryImplementationDetector {
+
+	@Substitute
+	private Set<BeanDefinition> findCandidateBeanDefinitions(ImplementationDetectionConfiguration config) {
+
+		/* Using the index instead of ClassPathScanningCandidateComponentProvider with pattern.
+		 * Not sure why components are not found via the index as it should be configured on
```

Comment:
@aclement Could you please give us your advice on that question raised by @christophstrobl ?

Reply:
I think that is worth an issue in its own right. I saw a similar thing whilst working with Josh Long - he was trying to use `ClassPathScanningCandidateComponentProvider` (IIRC) and it wasn't always able to use the components index - as if the combination of filters you set on it can cause it to not allow index usage.

Reply:
@christophstrobl Could you please create the issue suggested by Andy?

@aclement Do you think we should merge this PR as it is of should we wait the issue mention above to be fixed?

Reply:
@sdeleuze I would say this improves things, let's integrate it. We have the issue for improving things further.